### PR TITLE
Add filtering by date

### PR DESCRIPTION
**Pull request by aziz512**
Improved version of https://github.com/nycJSorg/angular-presentation/pull/451

---
**Default:**

image
---
**One bound set:**

image
---
**Two bounds set:**

image

### CODE REVIEW

**Comment by kirjs** [comment]
Please add screenshot for the context add fix the build

**Comment by AleksandrBu** [comment]
I would suggest to add some kind of choices for user:
1) UpToToday (default). Where dateFrom = 01.01.1970, dateTo = new Date()
2) UpToDate. Where dateFrom = 01.01.1970, dateTo - let user to set in special input field
3) period. Where dateFrom = new Date() - 1 month, dateTo = new Date

And that's would be cool add reset button) and set default values like, for example, in 2) above

How do you think?

**Comment by aziz512** [comment]
@AleksandrBu 
Currently, a blank value (mm/dd/yyy seen on screenshot) means a lack of limit.
This way, blank value for the dateFrom means 01.01.1970 and blank value for dateTo means no upper limit.
That being said, I think presets like UpToDate will be redundant. Nevertheless, I see the value in a clear (reset) button, and I'll add it soon.

File: apps/codelab/src/app/codelabs/extra/feedback-page/date-range/date-range.component.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { BehaviorSubject, combineLatest } from 'rxjs';
+import { debounceTime } from 'rxjs/operators';
+import { FormGroup, FormControl } from '@angular/forms';
+
+@Component({
```
Comment (kirjs) [inline apps/codelab/src/app/codelabs/extra/feedback-page/date-range/date-range.component.ts]: OnPush?

File: apps/codelab/src/app/codelabs/extra/feedback-page/date-range/date-range.component.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<form [formGroup]="rangeForm">
+  <span>
+    from: <input type="date" formControlName="from" />
```
Comment (alex-bu-93) [inline apps/codelab/src/app/codelabs/extra/feedback-page/date-range/date-range.component.html]: Let's name the variable as 'dateFrom'?

**Comment by alex-bu-93** [comment]
Aziz, may be add "No data to show" placeholder if there is no issues after filtering? How do you think?
image

**Pull request merged**